operators/src/pro/cache/shared_cache.rs

```diff
@@ -966,28 +979,46 @@ where
     /// Insert a cachable element for a given query. The query has to be inserted first.
     /// The element is inserted into the landing zone and only moved to the cache when the query is finished.
     /// If the landing zone is full or the element size would cause the landing zone size to overflow, the caching of the query is aborted.
-    async fn insert_query_element<S: CacheElementSubType<CacheElementType = C>>(
+    async fn insert_query_element(
         &self,
         key: &CanonicOperatorName,
         query_id: &QueryId,
         landing_zone_element: C,
     ) -> Result<(), CacheError> {
+        let storeable_element = if log_enabled!(log::Level::Trace) {
+            let element_size = landing_zone_element.byte_size();
+            let storeable_element =
+                crate::util::spawn_blocking(|| landing_zone_element.into_stored_element())
+                    .await
+                    .map_err(|_| CacheError::BlockingElementConversion)?;
+            let storeable_element_size = storeable_element.byte_size();
+            tracing::trace!(
+                "Inserting element into landing zone for query {:?} on operator {}. Element size: {} bytes, storable element size: {} bytes, ratio: {}",
+                query_id,
+                key,
+                element_size,
+                storeable_element_size,
+                storeable_element_size as f64 / element_size as f64
+            );
+            storeable_element
+        } else {
+            crate::util::spawn_blocking(|| landing_zone_element.into_stored_element())
```

Comment:
can you avoid duplicating code here? you have the spawn_blocking twice.

Reply:
[2df1dd8](https://github.com/geo-engine/geoengine/pull/848/commits/2df1dd88bec22458d7aacc2e33dd6eddd865b19b)
+
[59de662](https://github.com/geo-engine/geoengine/pull/848/commits/59de6629c6ee4064a1b574d91b566410a2629891) (Copilot forgot the to increase the index)